lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -441,6 +1975,244 @@ bool StackAnalysis::runOnModule(Module &M) {
+  }
+
+  BasicBlockNode FakeNode{ nullptr };
+  auto *RootNode = CG.addNode(FakeNode);
+  CG.setEntryNode(RootNode);
+  for (const auto &Node : EntryBlocks)
+    RootNode->addSuccessor(Node);
+
+  if (StackAnalysisLog.isEnabled()) {
+    std::error_code EC;
+    raw_fd_ostream Out(M.getModuleIdentifier() + "_CG.dot",
+                       EC,
+                       llvm::sys::fs::OF_None);
+    revng_assert(!EC);
+    llvm::WriteGraph(Out, &CG);
+    revng_log(StackAnalysisLog, "Call graph written on disk");
+    Out.close();
+  }
+
+  for (auto *Node : llvm::post_order(&CG)) {
+    if (Node->hasPredecessors())
+      CFEPQueue.insert(Node);
+  }
+
+  // Collect all the ABI registers
+  std::vector<llvm::GlobalVariable *> ABIRegisters;
+  for (auto *CSV : GCBI.abiRegisters())
+    if (CSV && !(GCBI.isSPReg(CSV)))
+      ABIRegisters.emplace_back(CSV);
+
+  std::optional<int64_t> FSO;
+  switch (GCBI.arch()) {
+  case llvm::Triple::x86: {
+    FSO = 4;
+    break;
+  }
+  case llvm::Triple::x86_64: {
+    FSO = 8;
+    break;
+  }
+  default: {
+    FSO = 0;
+    break;
+  }
+  }
+
+  // Default-constructed cache summary for indirect calls
+  FunctionSummary DefaultSummary(model::FunctionType::Values::Regular,
+                                 { ABIRegisters.begin(), ABIRegisters.end() },
+                                 ABIAnalyses::ABIAnalysesResults(),
+                                 {},
+                                 FSO,
+                                 nullptr);
+  FunctionAnalysisResults Properties(std::move(DefaultSummary));
+
+  // Open streams for dumping results
+  std::unique_ptr<raw_fd_ostream> OutputIBI;
+  if (IndirectBranchInfoSummaryPath.getNumOccurrences() == 1) {
+    std::remove(IndirectBranchInfoSummaryPath.c_str());
+    std::error_code EC;
+    OutputIBI = std::make_unique<raw_fd_ostream>(IndirectBranchInfoSummaryPath,
+                                                 EC,
+                                                 llvm::sys::fs::OF_Append);
+    revng_assert(!EC);
+
+    *OutputIBI << "name,ra,fso,address";
+    for (const auto &Reg : ABIRegisters)
+      *OutputIBI << "," << Reg->getName();
+    *OutputIBI << "\n";
+  }
+
+  std::unique_ptr<raw_fd_ostream> OutputAAWriter;
+  if (AAWriterPath.getNumOccurrences() == 1) {
+    std::remove(AAWriterPath.c_str());
+    std::error_code EC;
+    OutputAAWriter = std::make_unique<raw_fd_ostream>(AAWriterPath,
+                                                      EC,
+                                                      llvm::sys::fs::OF_Append);
+    revng_assert(!EC);
+  }
+
+  using CFEPA = CFEPAnalyzer<FunctionAnalysisResults>;
+  CFEPA Analyzer(M,
+                 &GCBI,
+                 Properties,
+                 ABIRegisters,
+                 PreHookMarker,
+                 PostHookMarker,
+                 RetHookMarker,
+                 UnexpectedPCMarker,
+                 *OutputIBI,
+                 *OutputAAWriter);
```

Comment:
Review each constructor argument. If the argument is only used within the class, move it into the class.
Most of them look like they'd benefit from that.

Reply:
You mean instantiating the object in `CFEPA` constructor (instead of passing it as argument), right?

Reply:
Yep.